app/controllers/verify/authentications_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Verify
+  class AuthenticationsController < ApplicationController
+    before_action :send_unstarted_claiments_to_the_start
+    skip_before_action :verify_authenticity_token, only: [:create]
+
+    # Page where a new Verify authentication request is generated and posted, as
+    # described here:
+    #
+    #   https://www.docs.verify.service.gov.uk/get-started/set-up-successful-verification-journey/#generate-an-authentication-request
+    def new
+      @verify_authentication_request = Verify::AuthenticationRequest.generate
+      session[:verify_request_id] = @verify_authentication_request.request_id
+    end
+
+    # Callback where Verify will POST the SAML response for the authentication
+    # attempt, as described here:
+    #
+    #   https://www.docs.verify.service.gov.uk/get-started/set-up-successful-verification-journey/#handle-a-response
+    def create
+      verify_authentication_response = Verify::ServiceProvider.new.translate_response(params["SAMLResponse"], session[:verify_request_id], "LEVEL_2")
+      current_claim.update!(claim_parameters_from_response(verify_authentication_response))
+      redirect_to claim_path("teacher-reference-number")
+    end
+
+    private
+
+    def claim_parameters_from_response(response)
```

Comment:
Should this be in the controller? Might be better in the model, or better still, a small service object

Reply:
Yep, I definitely think we should break out the processing of the verify responses into their own object(s), but I think we'll be in a better place to extract that object when we are handling the failure cases - the shape/name/responsibility of the abstraction(s) will be clearer at that point.

Reply:
Also, remind me to schedule a dev discussion about "service objects" at some point...

Reply:
Ha, yeah, we've had a brief chat about that. Just force of habit!

Reply:
But yeah, I think it's fine where it is for now